### DISCUSSION TITLE
Is there any reasonable way to simulate a narrower window, or dynamically change breakpoints?

### DISCUSSION THREAD

**Discussion by dvdsgl** [tailwindlabs/tailwindcss#5028] [Help] [answered]
Thank you for your amazing project!

We would like to display tailwind components inside of a `div` that simulates a browser window. Our users should be able to resize this `div` to see components resize responsively. Basically, it should work just like the tailwindui gallery:

![CleanShot 2021-07-20 at 20 28 22](https://user-images.githubusercontent.com/108197/126426052-183ba07a-ed96-4f2f-ac15-af1eb4fb9a89.gif)

tailwindui uses iframes to do this, but we tested that approach and it breaks JavaScript libraries that access `window` due to the way we portal our code. If container queries were widely supported (proof-of-concept in https://github.com/tailwindlabs/tailwindcss/discussions/4075), that would probably be the solution.

If we could dynamically adjust breakpoints, I think we could achieve the same interaction—simulating a narrower, dynamic window width. **Is there any reasonable way to do this?**

**Comment by olets** [comment]
There are some approaches at https://stackoverflow.com/q/12251750/1241736

**Comment by dvdsgl** [comment, marked as answer]
We accomplished this by adding `responsive-{sm,md,lg,xl,2xl}` variants to https://github.com/ben-rogerson/twin.macro, and manually setting these classes on the pseudo browser, to simulate dynamic responsiveness within a container:

```patch
diff --git a/node_modules/twin.macro/macro.js b/node_modules/twin.macro/macro.js
index 8b28d1d..8d1e40b 100644
--- a/node_modules/twin.macro/macro.js
+++ b/node_modules/twin.macro/macro.js
@@ -2690,6 +2690,14 @@ var variantConfig = function (ref) {
    'group-hocus': function (variantData) { return prefixDarkLightModeClass('.group:hover &, .group:focus &', variantData); },
    'group-active': function (variantData) { return prefixDarkLightModeClass('.group:active &', variantData); },
    'group-visited': function (variantData) { return prefixDarkLightModeClass('.group:visited &', variantData); },
    +  
    +  'responsive-sm': function (variantData) { return prefixDarkLightModeClass('.responsive-sm &', variantData); },
    +  'responsive-md': function (variantData) { return prefixDarkLightModeClass('.responsive-md &', variantData); },
    +  'responsive-lg': function (variantData) { return prefixDarkLightModeClass('.responsive-lg &', variantData); },
    +  'responsive-xl': function (variantData) { return prefixDarkLightModeClass('.responsive-xl &', variantData); },
    +  'responsive-2xl': function (variantData) { return prefixDarkLightModeClass('.responsive-2xl &', variantData); },
    +  
    // Media types
    print: '@media print',
    screen: '@media screen',
```
